tests/core/test_processor.py

```diff
@@ -241,15 +254,198 @@
+
+    assert reminder_cancelled.cancels_job_with_name(job_name, sender_id)
+    assert not reminder_cancelled.cancels_job_with_name(job_name.upper(), sender_id)
+
+
+async def test_reminder_cancelled_cancels_job_with_name_special_name(
+    default_channel: CollectingOutputChannel, default_processor: MessageProcessor
+):
+    sender_id = "][]][xy,,=+2f'[:/;  >]<0d]A[e_,02"
+    name = "wkjbgr,34(,*&%^^&*(OP#LKMN V#NF# # #R"
+
+    reminder = ReminderScheduled(
+        intent="greet", trigger_date_time=datetime.datetime.now(), name=name
+    )
+    job_name = reminder.scheduled_job_name(sender_id)
+    reminder_cancelled = ReminderCancelled(name)
+
+    assert reminder_cancelled.cancels_job_with_name(job_name, sender_id)
+    assert not reminder_cancelled.cancels_job_with_name(job_name.upper(), sender_id)
+
+
+async def cancel_reminder_and_check(
+    tracker, default_processor, reminder_canceled_event, num_jobs_before, num_jobs_after
+):
+    # cancel the sixth reminder
+    tracker.update(reminder_canceled_event)
+
+    # check that the jobs were added
+    assert len((await jobs.scheduler()).get_jobs()) == num_jobs_before
+
+    await default_processor._cancel_reminders(tracker.events, tracker)
+
+    # check that only one job was removed
+    assert len((await jobs.scheduler()).get_jobs()) == num_jobs_after
+
+
+def default_reminder_list():
+    return [
+        ReminderScheduled("greet", datetime.datetime.now(), kill_on_user_message=False),
+        ReminderScheduled(
+            intent="greet",
+            entities=[{"entity": "name", "value": "Jane Doe"}],
+            trigger_date_time=datetime.datetime.now(),
+            kill_on_user_message=False,
+        ),
+        ReminderScheduled(
+            intent="default",
+            entities=[{"entity": "name", "value": "Jane Doe"}],
+            trigger_date_time=datetime.datetime.now(),
+            kill_on_user_message=False,
+        ),
+        ReminderScheduled(
+            intent="greet",
+            entities=[{"entity": "name", "value": "Bruce Wayne"}],
+            trigger_date_time=datetime.datetime.now(),
+            kill_on_user_message=False,
+        ),
+        ReminderScheduled(
+            "default", datetime.datetime.now(), kill_on_user_message=False
+        ),
+        ReminderScheduled(
+            "default",
+            datetime.datetime.now(),
+            kill_on_user_message=False,
+            name="special",
+        ),
+    ]
+
+
+async def test_reminder_cancelled_by_name(
+    default_channel: CollectingOutputChannel, default_processor: MessageProcessor
+):
+    reminders = default_reminder_list()
+    sender_id = uuid.uuid4().hex
+    tracker = default_processor.tracker_store.get_or_create_tracker(sender_id)
+    for reminder in reminders:
+        tracker.update(UserUttered("test"))
+        tracker.update(ActionExecuted("action_reminder_reminder"))
+        tracker.update(reminder)
+
+    default_processor.tracker_store.save(tracker)
```

Comment:
this part is duplicated multiple times. Might make sense to use a fixture for that (e.g. `tracker_with_scheduled_reminder`)